src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/services/OffenceCodeLookupService.kt

```diff
@@ -4,103 +4,46 @@ import org.springframework.stereotype.Service
-  }
+  fun getCommittedOnOwnNomisOffenceCodes(offenceCode: Int, isYouthOffender: Boolean): String = offenceDetails(offenceCode, isYouthOffender)?.nomisCode ?: OffenceCodeDefaults.DEFAULT_NOMIS_ID
 
-  fun getNotCommittedOnOwnNomisOffenceCode(offenceCode: Int, isYouthOffender: Boolean): String {
-    return offenceDetails(offenceCode, isYouthOffender)?.notCommittedOnOwnNomisCode ?: OffenceCodeDefaults.DEFAULT_NOMIS_ID
-  }
+  fun getNotCommittedOnOwnNomisOffenceCode(offenceCode: Int, isYouthOffender: Boolean): String = offenceDetails(offenceCode, isYouthOffender)?.notCommittedOnOwnNomisCode ?: OffenceCodeDefaults.DEFAULT_NOMIS_ID
 
-  fun getParagraphCode(offenceCode: Int, isYouthOffender: Boolean): String {
-    return offenceDetails(offenceCode, isYouthOffender)?.paragraph?.paragraphCode ?: OffenceCodeDefaults.DEFAULT_PARAGRAPH_DATA
-  }
+  fun getParagraphCode(offenceCode: Int, isYouthOffender: Boolean): String = offenceDetails(offenceCode, isYouthOffender)?.paragraphCode ?: OffenceCodeDefaults.DEFAULT_PARAGRAPH_DATA
 
-  fun getParagraphNumber(offenceCode: Int, isYouthOffender: Boolean): String {
-    return offenceDetails(offenceCode, isYouthOffender)?.paragraph?.paragraphNumber ?: OffenceCodeDefaults.DEFAULT_PARAGRAPH_DATA
-  }
+  fun getParagraphNumber(offenceCode: Int, isYouthOffender: Boolean): String =
+    offenceDetails(offenceCode, isYouthOffender)?.paragraphNumber ?: OffenceCodeDefaults.DEFAULT_PARAGRAPH_DATA
 
   fun getParagraphDescription(offenceCode: Int, isYouthOffender: Boolean): String {
-    return offenceDetails(offenceCode, isYouthOffender)?.paragraph?.paragraphDescription ?: OffenceCodeDefaults.DEFAULT_PARAGRAPH_DATA
+    val offenceCodeDetails = offenceDetails(offenceCode, isYouthOffender) ?: return Descriptions.DEFAULT.description
+    return offenceCodeParagraphs.getParagraphDescription(offenceCodeDetails.nomisCode)
   }
 
   private fun offenceDetails(offenceCode: Int, isYouthOffender: Boolean): OffenceCodeDetails? =
     when (isYouthOffender) {
-      true -> { youthOffenceDetailsByCode[offenceCode] }
-      false -> { adultOffenceDetailsByCode[offenceCode] }
-    }
-}
-
-class OffenceCodeLookupBuilder {
-  private var allCodeBuilders: MutableList<OffenceCodesBuilder> = mutableListOf()
-  private var nomisPrefix: String? = null
-
-  fun nomisPrefix(prefix: String): OffenceCodeLookupBuilder {
-    nomisPrefix = prefix
-    return this
-  }
-
-  fun code(code: Int): OffenceCodesBuilder {
-    val codeLookupBuilder = OffenceCodesBuilder(this, code, nomisPrefix!!)
-    allCodeBuilders.add(codeLookupBuilder)
-    return codeLookupBuilder
-  }
-
-  fun build(): Map<Int, OffenceCodeDetails> {
-    return allCodeBuilders.map { it.build() }.groupBy { it.offenceCode }
-      .mapValues {
-        if (it.value.size == 1) it.value[0] else throw RuntimeException("No single value provided for key ${it.key}")
+      true -> {
+        val res = youthOffenceDetailsByCode[offenceCode]
```

Comment:
I think you could refactor this and the other branches' logic to
```
youthOffenceDetailsByCode[offenceCode] ?.let {
  OffenceCodeDetails(offenceCode = offenceCode, paragraphNumber = it.first, nomisCode = it.second, notCommittedOnOwnNomisCode = it.third)
}
```

Reply:
updated